metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/TaxonomyTree.java

```diff
@@ -0,0 +1,149 @@
+    @JoinColumn(name = "class_id", referencedColumnName = "taxonomyId")
+    @OneToOne
+    private Taxonomy taxonomyClass;
+
+    @ApiModelProperty(position = 11, dataType = "java.lang.String", example = "[Url1, Url2]")
+    @JsonProperty
+    @OneToMany(fetch = FetchType.EAGER)
+    @JoinTable(name = "taxonomyTreeSubspecies", joinColumns = @JoinColumn(referencedColumnName = "species_id"),
+            inverseJoinColumns = @JoinColumn(referencedColumnName = "taxonomyId"))
+    private List<Taxonomy> taxonomySubSpecieses = new ArrayList<>();
+
+    public TaxonomyTree() {
+    }
+
+    public TaxonomyTree(Taxonomy taxonomySpecies, Taxonomy taxonomyGenus, Taxonomy taxonomyOrder, Taxonomy taxonomyClass) {
+        this.taxonomySpecies = taxonomySpecies;
+        this.taxonomyGenus = taxonomyGenus;
+        this.taxonomyOrder = taxonomyOrder;
+        this.taxonomyClass = taxonomyClass;
+    }
+
+    public Taxonomy getTaxonomySpecies() {
+        return taxonomySpecies;
+    }
+
+    public void setTaxonomySpecies(Taxonomy taxonomySpecies) {
+        this.taxonomySpecies = taxonomySpecies;
+    }
+
+    @ApiModelProperty(position = 1, readOnly = true)
+    public Long getTaxonomySpeciesId() {
+        return taxonomySpecies.getTaxonomyId();
+    }
+
+    public Taxonomy getTaxonomyGenus() {
+        return taxonomyGenus;
+    }
+
+    public void setTaxonomyGenus(Taxonomy taxonomyGenus) {
+        this.taxonomyGenus = taxonomyGenus;
+    }
+
+    @ApiModelProperty(position = 3, readOnly = true)
+    public Long getTaxonomyGenusId() {
+        return (taxonomyGenus == null) ? -1 : taxonomyGenus.getTaxonomyId();
+    }
+
+    public Taxonomy getTaxonomyOrder() {
+        return taxonomyOrder;
+    }
+
+    public void setTaxonomyOrder(Taxonomy taxonomyOrder) {
+        this.taxonomyOrder = taxonomyOrder;
+    }
+
+    @ApiModelProperty(position = 5, readOnly = true)
+    public Long getTaxOrderId() {
+        return (taxonomyOrder == null) ? -1 : taxonomyOrder.getTaxonomyId();
+    }
+
+    public Taxonomy getTaxonomyClass() {
+        return taxonomyClass;
+    }
+
+    public void setTaxonomyClass(Taxonomy taxonomyClass) {
+        this.taxonomyClass = taxonomyClass;
+    }
+
+    @ApiModelProperty(position = 7, readOnly = true)
+    public Long getTaxonomyClassId() {
+        return (taxonomyClass == null) ? -1 : taxonomyClass.getTaxonomyId();
+    }
+
+    @ApiModelProperty(position = 8, readOnly = true)
+    public List<Long> getTaxonomySubSpeciesesIds() {
+        return this.getTaxonomySubSpecieses().parallelStream().map(taxonomy -> taxonomy.getTaxonomyId())
+                .collect(Collectors.toList());
+    }
+
+    public List<Taxonomy> getTaxonomySubSpecieses() {
+        return taxonomySubSpecieses;
+    }
+
+    public void setTaxonomySubSpecieses(List<Taxonomy> taxonomySubSpecieses) {
+        this.taxonomySubSpecieses = taxonomySubSpecieses;
+    }
```

Comment:
The getters/setters are getting really repetitive, maybe we'll want to use something like this https://projectlombok.org/features/GetterSetter to get rid of this?

Reply:
Lombak is good to reduce the boilerplate code, maybe we can try in other ticket to check the vulnerabilities. we used Lombok long before and we faced issues in check style source code tools and was dependent on IDEs. but those won't be a problem now I guess.

Reply:
OK, then we'll need to create a ticket in the backlog for it

Reply:
Ticket created https://www.ebi.ac.uk/panda/jira/browse/T2D-349, resolving discussion